File: 0997-find-the-town-judge/0997-find-the-town-judge.java
class Solution {
    public int findJudge(int n, int[][] trust) {
        if(trust.length == 0 && n==1) return 1;
        int[] a = new int[n+1];
        int judge = -1;
        for(int i = 0; i < trust.length; i++){
            a[trust[i][1]]++;
            if(a[trust[i][1]] == n-1){
                judge = trust[i][1];
            }
        }
        for(int i = 0; i < trust.length; i++){
            if(trust[i][0] == judge) return -1;
        }
        return judge;
    }
}

File: 0997-find-the-town-judge/0997-find-the-town-judge.py
class Solution:
    def findJudge(self, n: int, trust: List[List[int]]) -> int:
        indegree = [0 for _ in range(n+1)]
        outdegree = [0 for _ in range(n+1)]
        
        for i in range(len(trust)):
            indegree[trust[i][0]] += 1
            outdegree[trust[i][1]] += 1
            
        #print(indegree)
        #print(outdegree)
        for i in range(1, n+1):
            if indegree[i] == 0 and outdegree[i] == n-1:
                return i
        return -1